internal/kargo/kargo_test.go

```diff
@@ -37,6 +37,17 @@ func TestNewPromotion(t *testing.T) {
 					Name:      "test",
 					Namespace: "kargo-demo",
 				},
+				Spec: kargoapi.StageSpec{
```

Comment:
Smallest of nits (higher up, because it's too far from the change to comment on): Do we want to consider renaming `TestNewPromotion()` to `TestPromotionBuilder()`?

Reply:
Oh... actually, I see `TestPromotionBuilder_Build` already exists.

Reply:
So now I wonder if there isn't some overlap between these two test funcs?

Reply:
Ah great catch, this is an old test that can just be removed but continued to work because I wrote something that was... backward compatible.